hazelcast-jet-sql/src/main/java/com/hazelcast/jet/sql/impl/schema/JetTableFunction.java

```diff
@@ -0,0 +1,181 @@
+/*
+ * Copyright (c) 2008-2020, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.jet.sql.impl.schema;
+
+import com.hazelcast.sql.impl.calcite.schema.HazelcastTable;
+import org.apache.calcite.rel.type.RelDataType;
+import org.apache.calcite.rel.type.RelDataTypeComparability;
+import org.apache.calcite.rel.type.RelDataTypeFactory;
+import org.apache.calcite.rel.type.RelDataTypeFamily;
+import org.apache.calcite.rel.type.RelDataTypeField;
+import org.apache.calcite.rel.type.RelDataTypePrecedenceList;
+import org.apache.calcite.rel.type.StructKind;
+import org.apache.calcite.schema.TableFunction;
+import org.apache.calcite.sql.SqlCollation;
+import org.apache.calcite.sql.SqlIdentifier;
+import org.apache.calcite.sql.SqlIntervalQualifier;
+import org.apache.calcite.sql.type.SqlTypeName;
+
+import java.nio.charset.Charset;
+import java.util.List;
+
+public abstract class JetTableFunction implements TableFunction {
+
+    @Override
+    public final RelDataType getRowType(RelDataTypeFactory typeFactory, List<Object> arguments) {
+        HazelcastTable table = toTable(arguments);
+        RelDataType rowType = table.getRowType(typeFactory);
+
+        return new JetFunctionRelDataType(table, rowType);
+    }
+
+    protected abstract HazelcastTable toTable(List<Object> arguments);
+
+    public HazelcastTable toTable(RelDataType rowType) {
+        return ((JetFunctionRelDataType) rowType).table();
+    }
+
+    private static final class JetFunctionRelDataType implements RelDataType {
```

Comment:
Is the only purpose of this class to be able to somehow pass the `HazelcastTable` object to place where the function is used?